src/routes/Swap.tsx

```diff
@@ -136,29 +122,57 @@ export default function Swap() {
     const handleSwap = async () => {
         if (canSwap()) {
             try {
+                setLoading(true);
                 const nodes = await state.mutiny_wallet?.list_nodes();
                 const firstNode = (nodes[0] as string) || "";
 
-                if (hasLsp()) {
-                    const new_channel = await state.mutiny_wallet?.open_channel(
-                        firstNode,
-                        undefined,
-                        amountSats()
-                    );
-
-                    setChannelOpenResult({ channel: new_channel });
+                if (isMax()) {
+                    console.log("sweeping all to channel");
+                    if (hasLsp()) {
```

Comment:
instead of all the if statements around `hasLsp()`, would be cleaner to do something like

```javascript
let peer = undefined;
if (hasLsp()) {
	peer = selectedPeer();
}
```

Reply:
thank you